### DISCUSSION TITLE
How to export @prisma/client to build files correctly

### DISCUSSION THREAD

**Discussion by octaviobarbosa** [prisma/prisma#10297] [Q&A] [answered]
Hello!

I have a project with all the prisma settings, schemas, models...etc and I need to export everything to build a folder to publish in an internal package where I use it in other projects.

index.ts
```javascript 
export * from "@prisma/client"
export * as helper from './connection'
export * from './utils'
```

how to export @prisma/client correcty ??

**Comment by austincrim** [comment, marked as answer]
Hey @octaviobarbosa 👋, thanks for the question.

I consulted with one of our engineers and he came up with a few pointers:
1. Generate the client in a dist folder by setting `output = "../dist"` in the `schema.prisma`.
1. The generated output folder comes with the bundled runtime, index, and binaries

If you want to add some extra logic to the package:
1. Set `output = "../dist/client"`
1. Create a `dist/index.js` where other logic can live, instantiate the client, export, etc.
From there you can directly import the client generated in `"../dist/client"`

One more important consideration, unless you know all the [Prisma engines](https://www.prisma.io/docs/concepts/components/prisma-engines) you will need in advance, you will need a `postinstall` script that runs a fresh `prisma generate`.

Let me know if you have more specific questions!

**Reply by octaviobarbosa** [reply]
Hey @austincrim thanks for the reply, it helped a lot!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
